tests/unit/checkpoint/test_zero_optimizer.py

```diff
@@ -544,6 +544,25 @@ def test_save_exclude_frozen_weights(self, tmpdir, zero_stage):
         trainable_param_names = set([n for n, p in model.named_parameters() if p.requires_grad])
         assert loaded_trainable_param_names == trainable_param_names
 
+        # Validate custom state_dict model
```

Comment:
Please make this a separate test, perhaps named `test_save_exclude_custom_frozen_weights`. It can copy the structure of the current test, but I think lines 521-545 would not be needed.

Reply:
Thanks, and I have fixed it!